src/lib/_loading.scss

```diff
@@ -99,3 +99,49 @@
         margin-left: 2px;
     }
 }
+
+@keyframes bp-spinner {
+    0% {
+        transform: translate(-50%, -50%) rotate(0deg);
+    }
+
+    100% {
+        transform: translate(-50%, -50%) rotate(360deg);
+    }
+}
+
+// Center spinner horizontally and vertically within container
+.bp-spinner {
+    bottom: 0;
+    left: 0;
+    margin: auto;
+    position: absolute;
+    right: 0;
+    top: 0;
```

Comment:
Using this absolute centering strategy: https://www.smashingmagazine.com/2013/08/absolute-horizontal-vertical-centering-css/